File: ports/go/src/core/global/bcscoretypes.go
package global

import "fmt"

// BcsGlobalColor defines standard core colors
type BcsGlobalColor int

const (
	BcsColorWhite BcsGlobalColor = iota
	BcsColorBlack
	BcsColorRed
	BcsColorGreen
	BcsColorBlue
	BcsColorTransparent
)

// BcsWarning handles internal framework warnings
func BcsWarning(msg string, args ...interface{}) {
	fmt.Printf("[BCS Warning] "+msg+"\n", args...)
}

// BcsCritical handles critical framework failures
func BcsCritical(msg string, args ...interface{}) {
	fmt.Printf("[BCS Critical] "+msg+"\n", args...)
}

// BcsAlignment defines vertical and horizontal layout constraints
type BcsAlignment int

const (
	AlignLeft BcsAlignment = 0x0001
	AlignRight BcsAlignment = 0x0002
	AlignHCenter BcsAlignment = 0x0004
	AlignJustify BcsAlignment = 0x0008
	AlignTop BcsAlignment = 0x0020
	AlignBottom BcsAlignment = 0x0040
	AlignVCenter BcsAlignment = 0x0080
	AlignCenter BcsAlignment = AlignVCenter | AlignHCenter
)

// BcsOrientation defines linear layout bounds
type BcsOrientation int

const (
	Horizontal BcsOrientation = 0x1
	Vertical BcsOrientation = 0x2
)

File: ports/rust/src/core/global/bcscoretypes.rs
// BcsGlobalColor defines standard core colors
pub enum BcsGlobalColor {
    White,
    Black,
    Red,
    Green,
    Blue,
    Transparent,
}

pub fn bcs_warning(msg: &str) {
    println!("[BCS Warning] {}", msg);
}

pub fn bcs_critical(msg: &str) {
    println!("[BCS Critical] {}", msg);
}

// BcsAlignment defines vertical and horizontal layout constraints
bitflags::bitflags! {
    pub struct BcsAlignment: u32 {
        const ALIGN_LEFT = 0x0001;
        const ALIGN_RIGHT = 0x0002;
        const ALIGN_HCENTER = 0x0004;
        const ALIGN_JUSTIFY = 0x0008;
        const ALIGN_TOP = 0x0020;
        const ALIGN_BOTTOM = 0x0040;
        const ALIGN_VCENTER = 0x0080;
        const ALIGN_CENTER = Self::ALIGN_VCENTER.bits | Self::ALIGN_HCENTER.bits;
    }
}

pub enum BcsOrientation {
    Horizontal = 0x1,
    Vertical = 0x2,
}
